source/docs/software/examples-tutorials/drivesim-tutorial/drivetrain-model.rst

```diff
@@ -0,0 +1,132 @@
+
+.. note:: It is very important to use SI units (i.e. meters and radians) when passing parameters in Java. In C++, the :ref:`units library <docs/software/basic-programming/cpp-units:The C++ Units Library>` can be used to specify any unit type.
+
+.. tabs::
+   .. code-tab:: java
+
+      // Create the simulation model of our drivetrain.
+      DifferentialDrivetrainSim m_driveSim = new DifferentialDrivetrainSim(
+        DCMotor.getNEO(2),       // 2 NEO motors on each side of the drivetrain.
+        7.29,                    // 7.29:1 gearing reduction.
+        7.5,                     // MOI of 7.5 kg m^2 (from CAD model).
+        60.0,                    // The mass of the robot is 60 kg.
+        Units.inchesToMeters(3), // The robot uses 3" radius wheels.
+        0.7112,                  // The track width is 0.7112 meters.
+
+        // The standard deviations for measurement noise:
+        // x and y:          0.001 m
+        // heading:          0.001 rad
+        // l and r velocity: 0.1   m/s
+        // l and r position: 0.005 m
+        VecBuilder.fill(0.001, 0.001, 0.001, 0.1, 0.1, 0.005, 0.005));
+
+   .. code-tab:: c++
+
+      #include <frc/simulation/DifferentialDrivetrainSim.h>
+
+      ...
+
+      // Create the simulation model of our drivetrain.
+      frc::sim::DifferentialDrivetrainSim m_driveSim{
+        DCMotor::GetNEO(2), // 2 NEO motors on each side of the drivetrain.
+        7.29,               // 7.29:1 gearing reduction.
+        7.5_kg_sq_m,        // MOI of 7.5 kg m^2 (from CAD model).
+        60_kg,              // The mass of the robot is 60 kg.
+        3_in,               // The robot uses 3" radius wheels.
+        0.7112_m,           // The track width is 0.7112 meters.
+
+        // The standard deviations for measurement noise:
+        // x and y:          0.001 m
+        // heading:          0.001 rad
+        // l and r velocity: 0.1   m/s
+        // l and r position: 0.005 m
+        {0.001, 0.001, 0.001, 0.1, 0.1, 0.005, 0.005}};
+
+Creating a ``DifferentialDrivetrainSim`` from Characterization Gains
+--------------------------------------------------------------------
+If at one point, you had access to your real robot, you may have performed :ref:`robot characterization <docs/software/wpilib-tools/robot-characterization/introduction:Introduction to Robot Characterization>` as part of setting up the trajectory tracking workflow outlined :ref:`here <docs/software/examples-tutorials/trajectory-tutorial/index:Trajectory Tutorial>`. These characterization gains can also be used to create a simulation model of your drivetrain and often yield results closer to real-world behavior than the method above.
```

Comment:
I think we should make this mandatory for if the user wants to follow along. They can't really skip this and the wording should reflect it.